### DISCUSSION TITLE
Autocompletion suggests duplicate literal in set argument

### DISCUSSION THREAD

**Discussion by JonathanPlasse** [microsoft/pylance-release#6232] [Enhancement]
## Environment data

-   Pylance version: 2024.7.104
-   OS and version: Ubuntu 20.4
-   Python version (& distribution if applicable, e.g. Anaconda): 3.10

## Code Snippet

```python
from typing import Literal

def f(_x: set[Literal["a", "b"]]) -> None:
    pass

f({"a", })
```

## Repro Steps

https://github.com/user-attachments/assets/029a476b-09fd-4e4c-878d-facc740f142e

1. Place the cursor after `f({"a",`  and trigger autocompletion

## Expected behavior

Autocompletion should not suggest "a" as it's already present in the set.

## Actual behavior

Autocompletion continues to suggest "a" even though it's already in the set.

**Comment by debonte** [comment]
This is correct because a `set[Literal["a", "b"]]` can be initialized to contain multiple instances of `"a"` and `"b"`, although the duplicates will be dropped.

But I see your point, that it seems strange that the completion suggestions would steer you in that direction.

**Comment by rchiodo** [comment]
Transferring to discussions for upvotes
